SCons/Tool/msvs.py

```diff
@@ -45,7 +45,9 @@
 import SCons.Warnings
 from SCons.Defaults import processDefines
 from SCons.compat import PICKLE_PROTOCOL
-from .MSCommon import msvc_exists, msvc_setup_env_once
+from .MSCommon import msvc_setup_env_tool, msvc_setup_env_once
+
+tool_name = os.path.splitext(os.path.basename(__file__))[0]
```

Comment:
can't we just hardcode?

Reply:
Sure.  I was avoiding hard-coding the name since the tool name is derived from filename.

Reply:
Got that, but it's just adds a level of complexity to understanding what's being passed rather than `tool='mslink'` for example.